lib/llm/conversation.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+module LLM
+  class Conversation
+    attr_accessor :thread
+
+    ##
+    # @param [Array<LLM::Message>] thread
+    #  An array of messages that form the conversation history
+    # @param [LLM::Provider] provider
+    #  A provider
+    def initialize(thread, provider)
```

Comment:
The order of arguments feels strange to me. I'd expect the provider to be first, the thread to be second.

Reply:
@0x1eef I totally agree! In fact i followed your earlier (messages, provider) order even though i was convinced the right order is the opposite :)